packages/contracts/contracts/CollateralManagerV2.sol

```diff
@@ -0,0 +1,506 @@
+pragma solidity >=0.8.0 <0.9.0;
+// SPDX-License-Identifier: MIT
+
+/*
+
+1. During submitBid, the collateral will be Committed (?) using the 'collateral validator'
+
+2. During acceptBid, the collateral gets bundled into the CollateralBundler which mints an NFT (the bundle) which then gets transferred into this contract 
+
+
+This collateral manager will only accept collateral bundles. 
+
+*/
+
+// Contracts
+import "@openzeppelin/contracts-upgradeable/utils/ContextUpgradeable.sol";
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+// Libraries
+import "@openzeppelin/contracts-upgradeable/utils/structs/EnumerableSetUpgradeable.sol";
+
+// Interfaces
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/IERC721Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC1155/IERC1155Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/IERC721ReceiverUpgradeable.sol";
+import "./interfaces/ICollateralManagerV2.sol";
+//import { Collateral, CollateralType, ICollateralEscrowV1 } from "./interfaces/escrow/ICollateralEscrowV1.sol";
+import "./interfaces/ITellerV2.sol";
+import "./bundle/TokenStore.sol";
+
+import "./bundle/interfaces/ICollateralBundle.sol";
+ 
+
+/*
+
+This contract is a token store which stores bundles.
+The bid id == the bundle id. 
+
+If the bundle exists and is owned by this contract, we know the collateral is held. 
+
+*/
+
+contract CollateralManagerV2 is
+    ContextUpgradeable,
+    TokenStore,
+    ICollateralManagerV2
+{
+    /* Storage */
+    using EnumerableSetUpgradeable for EnumerableSetUpgradeable.AddressSet;
+    ITellerV2 public tellerV2;
+
+    // bidIds -> collateralEscrow
+    //mapping(uint256 => address) public _escrows;
+
+    // bidIds -> collateralBundleId
+    mapping(uint256 => uint256) internal _collateralBundleIdForBid;
+
+    // bidIds -> collateralBundleInfo
+    //this just bridges the gap between submitBid and acceptBid
+    mapping(uint256 => ICollateralBundle.CollateralBundleInfo)
+        internal _committedBidCollateral;
+
+    /* Events */
+    event CollateralEscrowDeployed(uint256 _bidId, address _collateralEscrow);
+
+    //add events back !!
+    event CollateralCommitted(
+        uint256 _bidId,
+        CollateralType _type,
+        address _collateralAddress,
+        uint256 _amount,
+        uint256 _tokenId
+    );
+    event CollateralClaimed(uint256 _bidId);
```

Comment:
lets remove the `CollateralClaimed` event since we are already emitting `CollateralWithdrawn` from the `_withdraw` function
```suggestion
```